cpp/foxglove/src/server.cpp

```diff
@@ -1,11 +1,35 @@
 #include <foxglove-c/foxglove-c.h>
 #include <foxglove/server.hpp>
 
+namespace {
+
+bool hasAnyCallbacks(const foxglove::WebSocketServerCallbacks& callbacks) {
+  return callbacks.onSubscribe || callbacks.onUnsubscribe;
+}
+
+}  // anonymous namespace
+
 namespace foxglove {
 
 WebSocketServer::WebSocketServer(WebSocketServerOptions options)
-    : _impl(
-        foxglove_server_start(options.name.c_str(), options.host.c_str(), options.port),
+    : _callbacks(options.callbacks)
+    , _impl(
```

Comment:
That's a brutal initializer, is there a more readable way to do this in C++? I mean maybe this is idiomatic, I try not to remember.

What if you make a method on WebSocketServerOptions that returns foxglove_server_options?

Reply:
This got messy because I was trying to keep most of the initialization in the member-initializer-list, but I could just give up on that and put it in the constructor body...